VestingInstance.sol

```diff
@@ -0,0 +1,273 @@
+        // solhint-disable-next-line max-line-length
+         require(vestingStartTime.add(vestingCliff) > block.timestamp, "Final time is before current time");
+
+        // transfer the ownership from the factory to the vesting owner
+        _transferOwnership(vestingContractOwner);
+
+        //make the vesting instance an account
+        getAccounts().setAccount("unique account name", 0x0, getAccounts().getWalletAddress(beneficiary)); // TODO: check unique account name, key and wallet address(that of the beneficiary ???)
+
+        _vestingPeriods =  vestingAmount.div(vestAmountPerPeriod);
+        _beneficiary = beneficiary;
+        _vestingAmount = vestingAmount;
+        _vestAmountPerPeriod = vestAmountPerPeriod;
+        _revocable = revocable;
+        _vestingPeriodSec = vestingPeriodSec;
+        _cliffStartTime = vestingStartTime.add(vestingCliff);
+        _vestingStartTime = vestingStartTime;
+        _refundDestination = refundDestination;
+        _revoker = revoker;
+    }
+
+    /**
+     * @notice Transfers available released tokens from the vesting back to beneficiary.
+     */
+    function withdraw() external onlyBeneficiary {
+        uint256 releasableAmount = _getReleasableAmount(block.timestamp);
+
+        require(releasableAmount > 0, "No unreleased tokens are due for withdraw");
+
+        _currentlyReleased = _currentlyReleased.add(releasableAmount);
+
+        getGoldToken().safeTransfer(_beneficiary, releasableAmount);
+
+        emit VestingWithdrawn(msg.sender, releasableAmount, block.timestamp);
+    }
+
+    /**
+     * @notice Allows only the revoker to revoke the vesting. Gold already vested
+     * remains in the contract, the rest is returned to the _refundDestination.
+     * @param revokeTime the revocation timestamp
+     * @dev revokeTime the revocation timestamp. If is less than the current block timestamp, it is set equal
+     */
+    function revoke(revokeTime) external onlyRevoker {
+        require(_revocable, "Revoking is not allowed");
+        require(!_revoked, "Vesting already revoked");
+
+        uint256 revokeTimestamp = revokeTime > block.timestamp ? revokeTime : block.timestamp;
+
+        uint256 balance = getGoldToken().balanceOf(address(this));
+        uint256 releasableAmount = _getReleasableAmount(revokeTimestamp);
+        uint256 refund = balance.sub(releasableAmount);
+
+        _revoked = true;
+        _revokeTime = revokeTimestamp;
+
+        getGoldToken().transfer(_refundDestination, refund);
+
+        emit VestingRevoked(msg.sender, _refundDestination, refund, _revokeTime);
+    }
+
+    /**
+     * @dev Calculates the amount that has already vested but hasn't been withdrawn (released) yet.
+     * @param timestamp the timestamp at which the calculate the releasable amount
+     */
+    function _getReleasableAmount(uint256 timestamp) private view returns (uint256) {
```

Comment:
I think, that a public function might prove useful, for client-side applications trying to figure out how much "unwithdrawn releasable amount" is there

Reply:
of coourse, changed